lms/djangoapps/courseware/views/views.py

```diff
@@ -393,9 +396,13 @@ def course_info(request, course_id):
 
         response = render_to_response('courseware/info.html', context)
         if store_upgrade_cookie:
+            cookie_value = course_id
+            if upgrade_cookie_name in request.COOKIES and course_id not in request.COOKIES[upgrade_cookie_name]:
```

Comment:
Yes, this is something we discussed with @marcotuts (and he was fine with it). Having a separate cookie per course id would be nicer, but unfortunately we'd have to store the course_id as part of the cookie name. If nobody is concerned about the length of the cookie name, I guess that's fine.

I was thinking the "right" long-term solution is to store a mapping in the cookie value of course_id -> expiration date. But we thought was overkill given that seeing the banner longer than intended has little, if any, downside.